src/RobotInterface/YarpImplementation/include/BipedalLocomotion/RobotInterface/YarpSensorBridgeImpl.h

```diff
@@ -1709,6 +1720,37 @@ struct YarpSensorBridge::Impl
             }
         }
 
+        if (controlBoardRemapperInterfaces.motors)
+        {
+            for (int j = 0; j < controlBoardRemapperMeasures.motorTemperatureUnordered.size(); j++)
+            {
+                ok &= controlBoardRemapperInterfaces.motors
+                        ->getTemperature(j, &(controlBoardRemapperMeasures.motorTemperatureUnordered[j]));
```

Comment:
As discussed F2F we can set a negative value in case of an error, btw if I'm not mistaken @MSECode designed an error code, we may just avoid to check the ok

Reply:
yes in icub-main, in the `getTemperature()` method if it is reading -5000 it will return false. In any case you can  use a negative value for your error. If it is different from -5000 you'll read it. here the line: https://github.com/robotology/icub-main/blob/4595c48eb71a0ce64d61ea309bea329fb0c8bff7/src/libraries/icubmod/embObjMotionControl/embObjMotionControl.cpp#L4806

Reply:
Given the comment https://github.com/ami-iit/bipedal-locomotion-framework/pull/813#discussion_r1511339493, I would check all the cases mentioned in the documentation https://icub-tech-iit.github.io/documentation/temperature_sensors/software/dataflow/#error-handling, plus the case return false. What do you think @GiulioRomualdi ?

Reply:
Hi @MSECode and @GiulioRomualdi ,
please see the comment [above](https://github.com/ami-iit/bipedal-locomotion-framework/pull/813#discussion_r1511339493). 
The value `-5000`is strictly related to the embObjMotioncontrol implementation. It hasn't be used outside it!! So you need to check the return value (true/false) only.

If you are not interested in checking the type of error you just skip the negative value, but, please, currently the negative error can give us some information on what's up on the joint j. so you can use in the logging, but skip in the computation/verification algorithm.

Reply:
Note that assuming that getTemperature is reporting measures in degree Celsius (it is not documented in https://www.yarp.it/latest/classyarp_1_1dev_1_1IMotor.html#details, but I guess this is the case), any value lower then −273.15 cannot be physical reached, so it can be safely assumed to be some kind of intentional or not error condition, for any possible future implementation of `yarp::dev::IMotor` .